src/services/faultTreeService.tsx

```diff
@@ -26,14 +26,19 @@ export const findAll = async (): Promise<FaultTree[]> => {
   }
 };
 
-export const find = async (faultTreeUri: string): Promise<FaultTree> => {
+export const find = async (faultTreeUri: string): Promise<{ result: FaultTree; reqProb?: any }> => {
   try {
     const fragment = extractFragment(faultTreeUri);
     const response = await axiosClient.get<FaultTree[]>(`/faultTrees/${fragment}`, {
       headers: authHeaders(),
     });
-
-    return JsonLdUtils.compactAndResolveReferences<FaultTree>(response.data, CONTEXT);
+    let reqProb;
+    const upperBound = response.data?.manifestingEvent?.supertypes?.[0]?.failureRate?.requirement?.upperBound;
```

Comment:
We have a standard way that is used everywhere how to:
1) retrieve data from the server
2) shape the data according to our needs

What is the motivation for not only returning the `result` but also the `reqProb` here? In this way, we will not have a single source of data as we would be able to access some value using different paths:
- `rootReqProb`
-  `ftaTree?.manifestingEvent?.supertypes?.[0]?.failureRate?.requirement?.upperBound`

Is there any advantage of polluting this code in such a way? How about using `ftaTree?.manifestingEvent?.supertypes?.[0]?.failureRate?.requirement?.upperBound` directly in useEffect(() method instead? Does it have a drawback?

Reply:
Maybe you can even define `reqProb = ftaTree?.manifestingEvent?.supertypes?.[0]?.failureRate?.requirement?.upperBound` before entering `useEffect()`